src/idb.filesystem.js

```diff
@@ -109,46 +109,31 @@ function resolveToFullPath_(cwdFullPath, path) {
 
   var relativePath = path[0] != DIR_SEPARATOR;
   if (relativePath) {
-    fullPath = cwdFullPath;
-    if (cwdFullPath != DIR_SEPARATOR) {
-      fullPath += DIR_SEPARATOR + path;
-    } else {
-      fullPath += path;
-    }
+    fullPath = cwdFullPath + DIR_SEPARATOR + path;
   }
 
-  // Adjust '..'s by removing parent directories when '..' flows in path.
+  // Normalize '.'s,  '..'s and '//'s.
   var parts = fullPath.split(DIR_SEPARATOR);
+  var finalParts = [];
   for (var i = 0; i < parts.length; ++i) {
     var part = parts[i];
-    if (part == '..') {
-      parts[i - 1] = '';
-      parts[i] = '';
+    if (part === '..') {
+      // Go up one level.
+      if (!finalParts.length)
+        throw Error('Invalid path');
+      finalParts.pop();
+    } else if (part === '.') {
+      // Skip over the current directory.
+    } else if (part !== '') {
+      // Eliminate sequences of '/'s as well as possible leading/trailing '/'s. 
+      finalParts.push(part);
     }
   }
-  fullPath = parts.filter(function(el) {
-    return el;
-  }).join(DIR_SEPARATOR);
-
-  // Add back in leading slash.
-  if (fullPath[0] != DIR_SEPARATOR) {
-    fullPath = DIR_SEPARATOR + fullPath;
-  } 
-
-  // Replace './' by current dir. ('./one/./two' -> one/two)
-  fullPath = fullPath.replace(/\.\//g, DIR_SEPARATOR);
-
-  // Replace '//' with '/'.
-  fullPath = fullPath.replace(/\/\//g, DIR_SEPARATOR);
```

Comment:
This would perform only one round of `//` -> `/` conversion, so `////` would only convert to `//`, but not further.